pkg/kademlia/kademlia_test.go

```diff
@@ -5,253 +5,221 @@ package kademlia
-	defer func(d []dht.DHT) {
-		for _, v := range d {
-			err := v.Disconnect()
-			assert.NoError(t, err)
-		}
-	}(dhts)
+func TestLookup(t *testing.T) {
+	lis, err := net.Listen("tcp", ":0")
+	assert.NoError(t, err)
+
+	srv, mns := newTestServer([]*proto.Node{&proto.Node{Id: "foo"}})
+	go srv.Serve(lis)
+	defer srv.Stop()
+
+	k := func() *Kademlia {
+		// make new identity
+		id, err := node.NewID()
+		assert.NoError(t, err)
+		id2, err := node.NewID()
+		assert.NoError(t, err)
+		// initialize kademlia
+		k, err := NewKademlia(id, []proto.Node{proto.Node{Id: id2.String(), Address: &proto.NodeAddress{Address: lis.Addr().String()}}}, lis.Addr().String())
+		assert.NoError(t, err)
+		return k
+	}()
 
 	cases := []struct {
-		k            *Kademlia
-		limit        int
-		expectedErr  error
-		restrictions []overlay.Restriction
+		k           *Kademlia
+		target      dht.NodeID
+		opts        lookupOpts
+		expected    *proto.Node
+		expectedErr error
 	}{
 		{
-			k:           newTestKademlia(t, "127.0.0.1", "6000", bootNode),
-			limit:       10,
+			k: k,
+			target: func() *node.ID {
+				id, err := node.NewID()
+				assert.NoError(t, err)
+				mns.returnValue = []*proto.Node{&proto.Node{Id: id.String(), Address: &proto.NodeAddress{Address: lis.Addr().String()}}}
+				return id
+			}(),
+			opts:        lookupOpts{amount: 5},
+			expected:    &proto.Node{},
 			expectedErr: nil,
 		},
+		{
+			k: k,
+			target: func() *node.ID {
+				id, err := node.NewID()
+				assert.NoError(t, err)
+				return id
+			}(),
+			opts:        lookupOpts{amount: 5},
+			expected:    nil,
+			expectedErr: NodeNotFound,
+		},
 	}
 
 	for _, v := range cases {
-		defer v.k.Disconnect()
-		ctx := context.Background()
-		err := v.k.ListenAndServe()
+		actual, err := v.k.lookup(context.Background(), v.target, v.opts)
 		assert.Equal(t, v.expectedErr, err)
-		time.Sleep(time.Second)
-		err = v.k.Bootstrap(ctx)
-		assert.NoError(t, err)
-
-		rt, err := v.k.GetRoutingTable(context.Background())
-
-		assert.NoError(t, err)
-		start := rt.Local().Id
+		if v.expected != nil {
+			assert.Equal(t, v.target.String(), actual.GetId())
+		} else {
+			assert.Nil(t, actual)
+		}
 
-		nodes, err := v.k.GetNodes(ctx, start, v.limit, v.restrictions...)
-		assert.Equal(t, v.expectedErr, err)
-		assert.Len(t, nodes, v.limit)
-		v.k.dht.Disconnect()
+		time.Sleep(1 * time.Second)
 	}
 
+	assert.NoError(t, os.Remove("kbucket.db"))
+	assert.NoError(t, os.Remove("nbucket.db"))
 }
 
-func TestFindNode(t *testing.T) {
-	t.Skip()
-	dhts, bootNode := bootstrapTestNetwork(t, "127.0.0.1", "5001")
-	defer func(d []dht.DHT) {
-		for _, v := range d {
-			err := v.Disconnect()
-			assert.NoError(t, err)
-		}
-	}(dhts)
+func TestBootstrap(t *testing.T) {
+	// m := map[string]bool{}
+	// id, err := node.NewID()
+	// assert.NoError(t, err)
+	// m[id.String()] = false
+	nodes := createTestNodes(20)
+	defer cleanup(nodes)
+
+	// var last *grpc.Server
```

Comment:
i would rather not check in commented out code. can we put it in a branch or something somewhere?

Reply:
Areed, this will not be commented out before merging.